src/otx/algo/modules/conv_module.py

```diff
@@ -319,25 +302,45 @@ def __init__(
         stride: int | tuple[int, int] = 1,
         padding: int | tuple[int, int] = 0,
         dilation: int | tuple[int, int] = 1,
-        norm_cfg: dict | None = None,
-        activation_callable: Callable[..., nn.Module] = nn.ReLU,
-        dw_norm_cfg: dict | None = None,
-        dw_activation_callable: Callable[..., nn.Module] | None = None,
-        pw_norm_cfg: dict | None = None,
-        pw_activation_callable: Callable[..., nn.Module] | None = None,
+        normalization: tuple[str, nn.Module] | None = None,
+        activation: nn.Module | None = nn.ReLU,
+        dw_normalization: tuple[str, nn.Module] | None = None,
+        dw_activation: nn.Module | None = None,
+        pw_normalization: tuple[str, nn.Module] | None = None,
+        pw_activation: nn.Module | None = None,
         **kwargs,
     ):
         super().__init__()
         if "groups" in kwargs:
             msg = "groups should not be specified in DepthwiseSeparableConvModule."
             raise ValueError(msg)
 
+        def update_num_features(
+            normalization: tuple[str, nn.Module] | None,
+            num_channels: int,
+        ) -> tuple[str, nn.Module] | None:
+            if normalization is None:
+                return normalization
+
+            if normalization[1].num_features != num_channels:
+                init_signature = inspect.signature(normalization[1].__init__)
+                init_parameters = {
+                    k: getattr(normalization[1], k) for k in init_signature.parameters if k in normalization[1].__dict__
+                }
+                num_channels_name = "num_channels" if isinstance(normalization[1], nn.GroupNorm) else "num_features"
+                init_parameters.update({num_channels_name: num_channels})
+                normalization[1].__init__(**init_parameters)
+            return build_norm_layer(normalization, num_features=in_channels)
+
         # if norm/activation config of depthwise/pointwise Conv2dModule is not
         # specified, use default config.
-        dw_norm_cfg = dw_norm_cfg or norm_cfg
-        dw_activation_callable = dw_activation_callable or activation_callable
-        pw_norm_cfg = pw_norm_cfg or norm_cfg
-        pw_activation_callable = pw_activation_callable or activation_callable
+        dw_normalization = dw_normalization or deepcopy(normalization)
+        dw_normalization = update_num_features(dw_normalization, in_channels)
```

Comment:
It's just a question. I think it's better to remove `normalization` from arguments and require user to give both `dw_normalization` and `pw_normalization`. How do you think about it?

Reply:
It is necessary when it is unclear whether to use `DepthwiseSeparableConvModule` or `Conv2dModule` like below
https://github.com/openvinotoolkit/training_extensions/blob/a5470fc3665228be8c3fde0bdc0728b11b26c057/src/otx/algo/common/backbones/cspnext.py#L166-L173